1. Added the classes Contact , User and Application Main class- Conta…

nave-en opened the pull request:
…ctApplication

2. In the User Class Added a function for validating the user details

nave-en committed:
1. Added the classes Contact , User and Application Main class- ContactApplication
2. In the User Class Added a function for validating the user details

Saijayavinoth commented on src/main/lib/User.java:
```diff
@@ -0,0 +1,34 @@
+        }
+
+        String passwordInHM = userCredentials.get(email);
+        if (password.equals(passwordInHM)) {
```
password nullable?

nave-en replied:
Added the null check and added trim condition as well

Saijayavinoth commented on src/main/lib/User.java:
```diff
@@ -0,0 +1,34 @@
+import java.util.HashMap;
+
+public class User {
+    private HashMap<String, String> userCredentials = new HashMap<>();
```
why mail in userCredentials?

nave-en replied:
Current implementation of user Login credentials based on the mail and password only

nave-en committed:
Updated the first round Code review comments
- By adding the trim function
Added the new functionalities
- Displaying all the contacts
- Add a new contact
- Display the specific contact
- Search the specific contact

Saijayavinoth commented on src/main/lib/Contact.java:
```diff
@@ -0,0 +1,93 @@
+import java.util.TreeMap;
+
+public class Contact {
+    private TreeMap<String, ArrayList<String>> contacts;
```
```suggestion
    private TreeMap<String, List<String>> contacts;
```

Saijayavinoth commented on src/main/lib/Contact.java:
```diff
@@ -0,0 +1,93 @@
+            ArrayList<String> phoneNumbers = new ArrayList<>();
+            phoneNumbers.add(phone);
+            contacts.put(name, phoneNumbers);
+        }
```
```suggestion
        contacts.computeIfAbsent(name, k -> new ArrayList())
            .add(phone);
```

nave-en committed:
Updated the Review comments
Added the new functionalities
1. Edit the contact name
2. Search by the contact number
3. Show the duplicate contacts
4. Merge the duplicate contacts
5. Delete the contact name
